test/common/init/manager_impl_test.cc

```diff
@@ -15,7 +15,14 @@ void expectUninitialized(const Manager& m) { EXPECT_EQ(Manager::State::Uninitial
 void expectInitializing(const Manager& m) { EXPECT_EQ(Manager::State::Initializing, m.state()); }
 void expectInitialized(const Manager& m) { EXPECT_EQ(Manager::State::Initialized, m.state()); }
 
-TEST(InitManagerImplTest, AddImmediateTargetsWhenUninitialized) {
+template <typename T> class InitManagerImplTest : public ::testing::Test {};
+
+using TargetTypes = ::testing::Types<ExpectableTargetImpl, ExpectableEagerTargetImpl>;
+
+TYPED_TEST_SUITE(InitManagerImplTest, TargetTypes);
```

Comment:
Neat, I hadn't seen this before. I'm not sure if you can totally get away with it, though, if you follow my suggestion of removing the initialization function from the eager target. I think the expectation is that the two target types should not behave the same in all cases.

Reply:
Right. Once the fn is removed from `EagerTargetImpl` the existing test cases don't fit. Will refactor